packages/mask/src/components/shared/WalletStatusBox/TransactionList.tsx

```diff
@@ -0,0 +1,197 @@
+    const { t } = useI18N()
+    const { Utils } = useWeb3State()
+    const { classes, theme } = useStyles()
+
+    const statusTextMap: Record<TransactionStatusType, string> = {
+        [TransactionStatusType.NOT_DEPEND]: t('recent_transaction_pending'),
+        [TransactionStatusType.SUCCEED]: t('recent_transaction_success'),
+        [TransactionStatusType.FAILED]: t('recent_transaction_failed'),
+        [TransactionStatusType.CANCELLED]: t('recent_transaction_cancelled'),
+    }
+
+    const web3 = useWeb3()
+    const { value: targetAddress } = useAsync(async () => {
+        if (tx.receipt?.contractAddress) return tx.receipt.contractAddress
+        if (tx.payload?.params?.[0].to) return tx.payload.params[0].to as string
+        const transaction = await web3.eth.getTransaction(tx.hash)
+        return transaction.to
+    }, [web3, tx])
+    const address = (targetAddress || '').toLowerCase()
+    const txData = tx.payload?.params?.[0].data as string | undefined
+    const { value: functionName } = useAsync(async () => {
+        return getContractFunctionName(txData)
+    }, [txData])
+
+    const handleClear = useCallback(() => {
+        onClear(tx)
+    }, [onClear, tx])
+
+    const domain = useMemo(() => getContractOwnerDomain(address), [address])
+
+    const [txStatus, setTxStatus] = useState(tx.status)
+
+    useEffect(() => {
+        return WalletMessages.events.transactionStateUpdated.on((data) => {
+            switch (data.type) {
+                case TransactionStateType.CONFIRMED:
+                    if (tx.hash === data.receipt.transactionHash) {
+                        setTxStatus(TransactionStatusType.SUCCEED)
+                    }
+                    break
+                case TransactionStateType.FAILED:
+                    if (tx.hash === data.receipt?.transactionHash) {
+                        setTxStatus(TransactionStatusType.FAILED)
+                    }
+            }
+        })
+    }, [tx.hash])
+
+    return (
+        <Grid container {...rest}>
+            <Grid item className={classes.cell} textAlign="left" md={4}>
+                <Stack overflow="hidden">
+                    <Typography
+                        className={classes.methodName}
+                        title={functionName || ''}
+                        variant="body1"
+                        component="strong">
+                        {functionName}
+                    </Typography>
+                    <Typography variant="body1" color={theme.palette.text.secondary}>
+                        {format(tx.at, 'yyyy.MM.dd hh:mm')}
+                    </Typography>
+                </Stack>
+            </Grid>
+            <Grid item className={classes.cell} flexGrow={1} md={4} justifyContent="right">
+                <Typography variant="body1">
+                    {address && isSameAddress(domain, address) ? Utils?.formatAddress?.(address, 4) : domain || address}
```

Comment:
So the `domain` is an address? If so, I think the name is easy to cause misunderstanding

Reply:
`domainOrAddress`